cmd/mountlib/rc.go

```diff
@@ -54,7 +65,7 @@ Eg
 	})
 }
 
-// rcMount allows the mount command to be run from rc
+// mountRc allows the mount command to be run from rc
 func mountRc(_ context.Context, in rc.Params) (out rc.Params, err error) {
 	mountPoint, err := in.GetString("mountPoint")
```

Comment:
for more complicated functions it is easiest just to write this for locking purposes

```go
    mountMu.Lock()
    defer mountMu.Unlock()
```

That will mean that the mutex is unlocked on function exit regardless. It is easy to audit and in go1.14 and later there is no performance penalty for using defer like this.

It is really super easy to mess locking up (I speak from lots of experience messing it up here!) so keeping it simple is best. I'd probably use this form on all your functions then you and I don't need to worry about it!

Reply:
Ahh.. This looks easier :p

Reply:
Shouldn't the mutex be released as soon as the reading, modification is finished?

Reply:
Also, shouldn't read/read operation be allowed? As they are nonconflicting.

Reply:
I'm curious